4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -58,11 +64,9 @@ public OperationResult RemoveAnimal(Animal animal)
             }
         }
 
-        public OperationResult GetAnimalFromID(int uniqueAnimalId)
+        public OperationResult GetAnimalFromID(Guid uniqueAnimalId)
         {
-            var result = from animal in Animals
-                         where animal.UniqueAnimalId == uniqueAnimalId
-                         select animal;
+            var result = _Animals.Where(a => a.UniqueAnimalId == uniqueAnimalId);
```

Comment:
Note that because you are checking for count, don't have the FirstorDefault, just do .First();

It mean if somehow the code changes in the future, .First() will fail as its not expecting default. FirstOrDefault totally has its place, but becareful because "default" for objects is null - passing null back can be easily more dangerous than an exception here.